### PR TITLE
Publishing: Auto close nexus staging repository in CI

### PR DESCRIPTION
**Pull request by martinbonnin**
Wildly untested 😅 . To be tested during the next release

### CODE REVIEW

File: build-logic/src/main/kotlin/io/openfeedback/extensions/ProjectExt.kt [hunk 1]
```diff
@@ -38,28 +44,57 @@ fun Project.publishIfNeededTaskProvider(): TaskProvider<Task> {
...
+    return try {
+        this.extensions.extraProperties["ossStagingRepositoryId"] as String?
     } catch (ignored: ExtraPropertiesExtension.UnknownPropertyException) {
         null
     }
-    if (url != null) {
-        return url
+}
+
+private fun Project.getOrCreateOssStagingRepoId(): String {
```
Comment (GerardPaligot) [inline build-logic/src/main/kotlin/io/openfeedback/extensions/ProjectExt.kt]: If X modules are executed in parallel and they check the nullability of `repoId` at the same time, what is the impact of `nexusStagingClient.createRepository` X times?

Comment (martinbonnin) [reply build-logic/src/main/kotlin/io/openfeedback/extensions/ProjectExt.kt]: > what is the impact of nexusStagingClient.createRepository X times?

It's bad! I'll add a lock

Comment (martinbonnin) [reply build-logic/src/main/kotlin/io/openfeedback/extensions/ProjectExt.kt]: done

**Comment by GerardPaligot** [comment]
I don't know if it'll work too :D but we'll test it at the next release! :)

**Comment by martinbonnin** [comment]
late-late-show-james-corden

**Pull request merged**